packages/ai/core/util/messages-store.ts

```diff
@@ -0,0 +1,348 @@
+import { generateId as generateIdFunction, parsePartialJson } from '.';
+import type {
+  ReasoningUIPart,
+  TextUIPart,
+  ToolInvocation,
+  ToolInvocationUIPart,
+  UIMessage,
+} from '../types';
+import { throttle } from './throttle';
+
+type SubscriberCallback = () => void;
```

Comment:
Suggest a different name e.g. `ChatStoreSubscriber` and switching to an object to have the opportunity for more fine-grained notifications in the future:

```
interface ChatStoreSubscriber {
  onChatMessagesChanged(...)
}
```